### ISSUE TITLE
Caught in a loop during Facebook signup

### ISSUE THREAD

**Issue by ghing** [denverfoundation/storybase#827]
@jwirfs-brock reports:

> I'm trying to do some help materials for Floodlight signup (because we have a 
> couple of workshops coming up), and I ran into some weirdness with the 
> Facebook signup. Here's what happened:
> - If I was NOT signed in to Facebook and I clicked the "Sign up with Facebook" button on the Sign Up page, http://floodlightproject.org/en/accounts/register/, I got redirected to a Facebook sign in page
> - Once I logged in at Facebook, I was redirected back to the Floodlight Sign In page, http://floodlightproject.org/en/accounts/login/#_=_
> - Once logged in to Facebook, clicking on the, "Sign up with Facebook" button just kept taking me back to the Sign In page

**Comment by ghing** [comment]
I wasn't able to replicate this when creating a new Floodlight account and was able to register for Floodlight using Facebook. I was also able to log in to the account using Facebook.  I'll test and and see if the issue is with associating an existing account with Facebook via the "Sign up with Facebook" registration flow.

**Comment by jwirfs-brock** [comment]
Note that I didn't have problems when I tried to associate my existing Floodlight account with my Facebook account. I only had problems when I tried to create a new Floodlight account with that Facebook account.

At that point, I had dissociated my existing Floodlight account with my Facebook account, but maybe it was still causing the problem?

**Comment by ghing** [comment]
I confirmed that I was able to create an account through the non-social registration flow using the email address associated with my Facebook account. I was then able to log in using Facebook without problems.

Finally, I was even able to log out of my account and log in with Facebook by clicking the "Sign up with Facebook" button instead of the correct "Sign in with Facebook" button.

@jwirfs-brock, can you try the Facebook registration/login flow after clearing your browser data (especially cookies) or running in private browsing mode?

Also, I'm having a bit of trouble understanding the history of the account registration in question. Does your existing Floodlight account (the one that was previously associated) have the same email address as your Facebook account? If so, the Facebook account will always be associated with the existing Floodlight account.

**Comment by jwirfs-brock** [comment]
@ghing I was just doing this to try to get some screenshots from the Facebook sign up process.

No, the Facebook account actually has a different email address associated with it than the Floodlight account. My Floodlight account was created with jbrock@piton.org. The Facebook account is associated with jordanwb@gmail.com.

I'll try it with clearing my browser data.

**Comment by jwirfs-brock** [comment]
@ghing Oops, I think I got that wrong. My Floodlight account is associated with the same email address as the Facebook account. Perhaps I'm running into a problem because of that.

Any ideas on how I can get some screenshots of the sign up via Facebook process, because I can't do it with my account?

**Comment by ghing** [comment]
@jwirfs-brock, let me know what you need and I can take them for you

**Comment by jwirfs-brock** [comment]
@ghing Thanks! Basically, I need a screen shot of every discrete step within the Facebook sign up process. For Twitter, there were two potential views you could see after clicking the "Sign up with Twitter" button (if you were already logged in to Twitter, and if you weren't), so I made sure to get both. Here's the sequence I got from the Twitter sign up process:

[1 - Not logged in screen]
![screen shot 2013-08-06 at 9 42 45 am](https://f.cloud.github.com/assets/1020518/924367/cd5f4170-ff65-11e2-9d80-431e8ff0f8b3.png)
[2 - Logged in screen]
![screen shot 2013-08-06 at 9 46 38 am](https://f.cloud.github.com/assets/1020518/924357/b22afcd2-ff65-11e2-86fa-9d3f60d0a96a.png)
[3 - Redirecting back to Floodlight screen]
![screen shot 2013-08-06 at 9 47 07 am](https://f.cloud.github.com/assets/1020518/924359/b5cf2da4-ff65-11e2-930e-50ccda4e255b.png)
[4 - Additional information from Floodlight screen]
![screen shot 2013-08-06 at 9 48 54 am](https://f.cloud.github.com/assets/1020518/924360/b8338e78-ff65-11e2-94c7-c410ef7b76d5.png)

Basically, I need something like that only with Facebook. I already have screen shots of what comes before and after this.

**Comment by ghing** [comment]
I sent @jwirfs-brock the needed screenshots and haven't been able to replicate her issues. I'm going to close this.
